backend/tests/userFlatStats.spec.js

```diff
@@ -0,0 +1,105 @@
+require('dotenv').config({ path: './.env' });
+const jwt = require('jsonwebtoken');
+const { CrudServer } = require('../src/server');
+const request = require('supertest');
+const { assert, expect } = require('chai');
+const User = require('../src/api/users/user.model');
+
+describe('CurrentUser test suite', () => {
+  let server;
+
+  before(async () => {
+    const crudServer = new CrudServer();
+    await crudServer.startForTest();
+    server = crudServer.server;
+  });
+
+  describe('GET /users/stats/flat', () => {
+    context('when bad token was provided', () => {
+      let response;
+
+      before(async () => {
+        response = await request(server)
+          .get('/api/v1/users/stats/flat')
+          .set('Authorization', 'Bearer bad_token');
+      });
+
+      it('should return 401 error', () => {
+        assert.equal(response.status, 401);
+      });
+    });
+    context('when stats not initialized', () => {
+      let response, userDoc;
+
+      before(async () => {
+        userDoc = await User.create({
+          username: 'Test2',
+          email: 'test2@email.com',
+          passwordHash: 'password_hash',
+        });
+
+        const token = jwt.sign({ id: userDoc._id }, process.env.JWT_SECRET, {
+          expiresIn: 2 * 24 * 60 * 60,
+        });
+        userDoc.tokens.push(token);
+        await userDoc.save();
+
+        response = await request(server)
+          .get('/api/v1/users/stats/flat')
+          .set('Authorization', `Bearer ${token}`);
+      });
+
+      after(async () => {
+        await User.deleteOne({ _id: userDoc._id });
+      });
+
+      it('should return 403 error', () => {
+        assert.equal(response.status, 403);
+      });
+    });
+
+    context('when good token was provided', () => {
+      let response, userDoc;
+
+      before(async () => {
+        userDoc = await User.create({
+          username: 'Test2',
+          email: 'test2@email.com',
+          passwordHash: 'password_hash',
+        });
+
+        const token = jwt.sign({ id: userDoc._id }, process.env.JWT_SECRET, {
+          expiresIn: 2 * 24 * 60 * 60,
+        });
+        userDoc.tokens.push(token);
+        userDoc.flatPrice = 1;
```

Comment:
Nope, you need to test that calculations went right. For instance, you need to write flatPrice equal N, squareMeters M etc. and then check in response that exact expected values were returned